In [29]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [30]:
accident = pd.read_csv('datasets\\accident_data.csv')

In [31]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [32]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [33]:
accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [34]:
accident.dtypes

Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

In [35]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [36]:
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'], dayfirst = True, errors = 'coerce')

In [37]:
accident.dtypes

Index                              object
Accident_Severity                  object
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                   object
District Area                      object
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions            object
Road_Type                          object
Urban_or_Rural_Area                object
Weather_Conditions                 object
Vehicle_Type                       object
dtype: object

<h1> Insight No.1 </h1>

<h1>What District Area has the most accidents?</h1>

In [38]:
accident['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h1> Birmingham has the most Accidents in United Kingdom from 2019-2022 with 13,491 Accidents</h1>

<h1> Insight No.2 </h1>

<h1> What Vehicle Type has the most accidents? </h1>

In [39]:
accident['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64

<h1> Car is the most involve vehicle type in accident with over 497,992 accidents and contributes 75% of all Accidents in United Kingdom</h1>

<h1> Insight No.3 </h1>

<h1> How does Wet Road Surface Condition contributes to Serious Accident Severity? </h1>

In [40]:
acc2 = accident[(accident['Road_Surface_Conditions'] == 'Wet') & (accident['Accident_Severity'] == 'Fatal')]

<h1> Insight No.4 </h1>

<h1> How does Area Types and Weather Conditions contributes to Number of Vehicles in Accident?</h1>

In [41]:
accident[(['Area Types'] == 'Urban') & (accident['Weather_Conditions'] == 'fine no high winds') & (accident['Number_of_Vehicles'] == '2')]

Empty DataFrame
Columns: [Index, Accident_Severity, Accident Date, Latitude, Light_Conditions, District Area, Longitude, Number_of_Casualties, Number_of_Vehicles, Road_Surface_Conditions, Road_Type, Urban_or_Rural_Area, Weather_Conditions, Vehicle_Type]
Index: []

<h1>Insight No.5 </h1>

<h1> What is the Relation of Road Type to the Accident Severity and Number of Casualties in Accident</h1>

<h1> UniVariate </h1>

<h1> Insight No.6 </h1>

In [42]:
accident['Year'] = accident['Accident Date'].dt.year
accident['Month'] = accident['Accident Date'].dt.month
accident['Day'] = accident['Accident Date'].dt.day
accident['DayOfWeek'] = accident['Accident Date'].dt.dayofweek

In [43]:
accident[accident['Year'] == 2021]

Index Accident_Severity Accident Date   Latitude  \
352706  200901BS70001           Serious    2021-01-01  51.512273   
352707  200901BS70002           Serious    2021-01-05  51.514399   
352708  200901BS70003            Slight    2021-01-04  51.486668   
352709  200901BS70004           Serious    2021-01-05  51.507804   
352710  200901BS70005           Serious    2021-01-06  51.482076   
...               ...               ...           ...        ...   
516241       2.01E+12            Slight    2021-11-12  55.358904   
516242       2.01E+12            Slight    2021-11-12  55.166976   
516247       2.01E+12            Slight    2021-12-01  55.115346   
516250       2.01E+12            Slight    2021-12-01  55.313433   
516253       2.01E+12            Slight    2021-12-09  55.115088   

              Light_Conditions           District Area  Longitude  \
352706                Daylight  Kensington and Chelsea  -0.201349   
352707                Daylight  Kensington and Chelsea  -0.199248   
352708                Daylight  Kensington and Chelsea  -0.179599   
352709                Daylight  Kensington and Chelsea  -0.203110   
352710   Darkness - lights lit  Kensington and Chelsea  -0.173445   
...                        ...                     ...        ...   
516241                Daylight   Dumfries and Galloway  -3.513667   
516242  Darkness - no lighting   Dumfries and Galloway  -3.389896   
516247  Darkness - no lighting   Dumfries and Galloway  -3.409426   
516250   Darkness - lights lit   Dumfries and Galloway  -3.454262   
516253  Darkness - no lighting   Dumfries and Galloway  -3.376962   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
352706                     1                   2                     Dry   
352707                    11                   2             Wet or damp   
352708                     1                   2                     Dry   
352709                     1                   2            Frost or ice   
352710                     1                   2                     Dry   
...                      ...                 ...                     ...   
516241                     1                   2             Wet or damp   
516242                     1                   1             Wet or damp   
516247                     1                   2            Frost or ice   
516250                     1                   1                    Snow   
516253                     2                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
352706      One way street               Urban     Fine no high winds   
352707  Single carriageway               Urban     Fine no high winds   
352708  Single carriageway               Urban     Fine no high winds   
352709  Single carriageway               Urban                  Other   
352710  Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
516241    Dual carriageway               Rural  Raining no high winds   
516242    Dual carriageway               Rural  Raining no high winds   
516247  Single carriageway               Rural                  Other   
516250    Dual carriageway               Rural  Snowing no high winds   
516253  Single carriageway               Rural     Fine no high winds   

                 Vehicle_Type    Year  Month   Day  DayOfWeek  
352706                    Car  2021.0    1.0   1.0        4.0  
352707  Taxi/Private hire car  2021.0    1.0   5.0        1.0  
352708  Taxi/Private hire car  2021.0    1.0   4.0        0.0  
352709  Motorcycle over 500cc  2021.0    1.0   5.0        1.0  
352710                    Car  2021.0    1.0   6.0        2.0  
...                       ...     ...    ...   ...        ...  
516241  Motorcycle over 500cc  2021.0   11.0  12.0        4.0  
516242                    Car  2021.0   11.0  12.0        4.0  
5162

In [44]:
df_2021 = accident[accident['Year'] == 2021]

In [45]:
df_2021

Index Accident_Severity Accident Date   Latitude  \
352706  200901BS70001           Serious    2021-01-01  51.512273   
352707  200901BS70002           Serious    2021-01-05  51.514399   
352708  200901BS70003            Slight    2021-01-04  51.486668   
352709  200901BS70004           Serious    2021-01-05  51.507804   
352710  200901BS70005           Serious    2021-01-06  51.482076   
...               ...               ...           ...        ...   
516241       2.01E+12            Slight    2021-11-12  55.358904   
516242       2.01E+12            Slight    2021-11-12  55.166976   
516247       2.01E+12            Slight    2021-12-01  55.115346   
516250       2.01E+12            Slight    2021-12-01  55.313433   
516253       2.01E+12            Slight    2021-12-09  55.115088   

              Light_Conditions           District Area  Longitude  \
352706                Daylight  Kensington and Chelsea  -0.201349   
352707                Daylight  Kensington and Chelsea  -0.199248   
352708                Daylight  Kensington and Chelsea  -0.179599   
352709                Daylight  Kensington and Chelsea  -0.203110   
352710   Darkness - lights lit  Kensington and Chelsea  -0.173445   
...                        ...                     ...        ...   
516241                Daylight   Dumfries and Galloway  -3.513667   
516242  Darkness - no lighting   Dumfries and Galloway  -3.389896   
516247  Darkness - no lighting   Dumfries and Galloway  -3.409426   
516250   Darkness - lights lit   Dumfries and Galloway  -3.454262   
516253  Darkness - no lighting   Dumfries and Galloway  -3.376962   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
352706                     1                   2                     Dry   
352707                    11                   2             Wet or damp   
352708                     1                   2                     Dry   
352709                     1                   2            Frost or ice   
352710                     1                   2                     Dry   
...                      ...                 ...                     ...   
516241                     1                   2             Wet or damp   
516242                     1                   1             Wet or damp   
516247                     1                   2            Frost or ice   
516250                     1                   1                    Snow   
516253                     2                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
352706      One way street               Urban     Fine no high winds   
352707  Single carriageway               Urban     Fine no high winds   
352708  Single carriageway               Urban     Fine no high winds   
352709  Single carriageway               Urban                  Other   
352710  Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
516241    Dual carriageway               Rural  Raining no high winds   
516242    Dual carriageway               Rural  Raining no high winds   
516247  Single carriageway               Rural                  Other   
516250    Dual carriageway               Rural  Snowing no high winds   
516253  Single carriageway               Rural     Fine no high winds   

                 Vehicle_Type    Year  Month   Day  DayOfWeek  
352706                    Car  2021.0    1.0   1.0        4.0  
352707  Taxi/Private hire car  2021.0    1.0   5.0        1.0  
352708  Taxi/Private hire car  2021.0    1.0   4.0        0.0  
352709  Motorcycle over 500cc  2021.0    1.0   5.0        1.0  
352710                    Car  2021.0    1.0   6.0        2.0  
...                       ...     ...    ...   ...        ...  
516241  Motorcycle over 500cc  2021.0   11.0  12.0        4.0  
516242                    Car  2021.0   11.0  12.0        4.0  
5162

In [46]:
df_2021['Road_Surface_Conditions'].value_counts().sum()

np.int64(66115)

In [47]:
df_2021['Road_Surface_Conditions'].value_counts()

Road_Surface_Conditions
Dry                     42641
Wet or damp             19869
Frost or ice             2543
Snow                      966
Flood over 3cm. deep       96
Name: count, dtype: int64

In [48]:
df_2021['Vehicle_Type'].value_counts().sum()

np.int64(66172)

In [49]:
df_2021['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      51745
Van / Goods 3.5 tonnes mgw or under       3359
Motorcycle over 500cc                     2441
Bus or coach (17 or more pass seats)      1683
Goods 7.5 tonnes mgw and over             1477
Motorcycle 125cc and under                1359
Taxi/Private hire car                     1104
Motorcycle 50cc and under                  784
Motorcycle over 125cc and up to 500cc      737
Other vehicle                              578
Goods over 3.5t. and under 7.5t            541
Agricultural vehicle                       181
Minibus (8 - 16 passenger seats)           161
Pedal cycle                                 20
Ridden horse                                 2
Name: count, dtype: int64

<h1>64% of 2021 Accidents happen in Dry Surface</h1>

<h1>78% of 2021 Accidents are Cars with Total of 51,745</h1>

<h1> Insight No.7 </h1>

In [50]:
accident[accident['Weather_Conditions'] == 'Fine no high winds']

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
3       200701BS66128           Serious           NaT  51.495478   
5       200701BS67159           Serious           NaT  51.497750   
6       200701BS67207           Serious    2019-09-05  51.501405   
7       200701BS67370             Fatal    2019-10-03  51.482260   
...               ...               ...           ...        ...   
660673  201091NM01190            Slight    2022-02-02  57.585152   
660674  201091NM01760            Slight           NaT  57.374005   
660675  201091NM01881            Slight           NaT  57.232273   
660676  201091NM01935            Slight           NaT  57.585044   
660677  201091NM01964           Serious           NaT  57.214898   

              Light_Conditions           District Area  Longitude  \
0        Darkness - lights lit  Kensington and Chelsea  -0.209082   
3                     Daylight  Kensington and Chelsea  -0.202731   
5                     Daylight  Kensington and Chelsea  -0.192556   
6                     Daylight  Kensington and Chelsea  -0.161577   
7        Darkness - lights lit  Kensington and Chelsea  -0.179486   
...                        ...                     ...        ...   
660673                Daylight                Highland  -3.748312   
660674                Daylight                Highland  -3.467828   
660675  Darkness - no lighting                Highland  -3.809281   
660676                Daylight                Highland  -3.862727   
660677  Darkness - no lighting                Highland  -3.823997   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
3                          1                   4                     Dry   
5                          2                   3                     Dry   
6                          1                   2                     Dry   
7                          3                   2                     Dry   
...                      ...                 ...                     ...   
660673                     2                   2             Wet or damp   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   

                 Road_Type Urban_or_Rural_Area  Weather_Conditions  \
0       Single carriageway               Urban  Fine no high winds   
3       Single carriageway               Urban  Fine no high winds   
5       Single carriageway               Urban  Fine no high winds   
6         Dual carriageway               Urban  Fine no high winds   
7       Single carriageway               Urban  Fine no high winds   
...                    ...                 ...                 ...   
660673  Single carriageway               Rural  Fine no high winds   
660674  Single carriageway               Rural  Fine no high winds   
660675  Single carriageway               Rural  Fine no high winds   
660676  Single carriageway               Rural  Fine no high winds   
660677  Single carriageway               Rural  Fine no high winds   

                                Vehicle_Type    Year  Month  Day  DayOfWeek  
0                                        Car  2019.0    6.0  5.0        2.0  
3       Bus or coach (17 or more pass seats)     NaN    NaN  NaN        NaN  
5                                        Car     NaN    NaN  NaN        NaN  
6        Van / Goods 3.5 tonnes mgw or under  2019.0    9.0  5.0        3.0  
7                                        Car  2019.0   10.0  3.0        3.0  
...                                      ...     ...    ...  ...        ...  
660673                                   Car  2022.0    2.0  2.0      

<h1> 79% of All Accidents Happen in Fine No High Winds Weather Condition with Total of 520885 Accidents. </h1>

<h1> Insight No.8 </h1>

In [51]:
 accident['Road_Type'].value_counts()

Road_Type
Single carriageway    492143
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64

In [52]:
accident['Urban_or_Rural_Area'].value_counts()

Urban_or_Rural_Area
Urban          421663
Rural          238990
Unallocated        11
Name: count, dtype: int64

<h1> 74% of all Accidents happen in Single Carriageway and 64% of all Accidents happen in Urban Area </h1>

<h1> Insight No.9 </h1>

<h1> Does accident increased from 2019 to 2020? </h1>

In [53]:
accident[accident['Year'] == 2019]

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
4       200701BS66837            Slight    2019-09-03  51.488576   
6       200701BS67207           Serious    2019-09-05  51.501405   
7       200701BS67370             Fatal    2019-10-03  51.482260   
...               ...               ...           ...        ...   
182097       2.01E+12            Slight    2019-11-07  55.212275   
182101       2.01E+12            Slight    2019-12-02  55.009139   
182102       2.01E+12            Slight    2019-12-03  54.993061   
182103       2.01E+12           Serious    2019-12-06  55.127391   
182104       2.01E+12            Slight    2019-12-07  54.988202   

              Light_Conditions           District Area  Longitude  \
0        Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                     Daylight  Kensington and Chelsea  -0.173647   
4        Darkness - lights lit  Kensington and Chelsea  -0.192487   
6                     Daylight  Kensington and Chelsea  -0.161577   
7        Darkness - lights lit  Kensington and Chelsea  -0.179486   
...                        ...                     ...        ...   
182097  Darkness - no lighting   Dumfries and Galloway  -3.413007   
182101                Daylight   Dumfries and Galloway  -3.319379   
182102                Daylight   Dumfries and Galloway  -3.131122   
182103   Darkness - lights lit   Dumfries and Galloway  -3.439964   
182104  Darkness - no lighting   Dumfries and Galloway  -3.122077   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
4                          1                   2                     Dry   
6                          1                   2                     Dry   
7                          3                   2                     Dry   
...                      ...                 ...                     ...   
182097                     1                   2                     Dry   
182101                     1                   1             Wet or damp   
182102                     1                   1            Frost or ice   
182103                     1                   2             Wet or damp   
182104                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
4                      NaN               Urban                    NaN   
6         Dual carriageway               Urban     Fine no high winds   
7       Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
182097    Dual carriageway               Rural     Fine no high winds   
182101  Single carriageway               Rural     Fine no high winds   
182102  Single carriageway               Rural     Fine no high winds   
182103  Single carriageway               Rural  Raining no high winds   
182104  Single carriageway               Rural     Fine no high winds   

                               Vehicle_Type    Year  Month  Day  DayOfWeek  
0                                       Car  2019.0    6.0  5.0        2.0  
1                                       Car  2019.0    7.0  2.0        1.0  
4                             Other vehicle  2019.0    9.0  3.0        1.0  
6       Van / Goods 3.5 tonnes mgw or under  2019.0    9.0  5.0        3.0  
7                                       Car  2019.0   10.0  3.0        3.0  
...                                     ...     ...    ...  ...        ...  
182097                                  C

In [54]:
accident[accident['Year'] == 2020]

Index Accident_Severity Accident Date   Latitude  \
182120  200801BS70001           Serious    2020-01-02  51.491521   
182121  200801BS70002            Slight    2020-01-06  51.484909   
182122  200801BS70003           Serious    2020-01-12  51.509965   
182123  200801BS70004            Slight    2020-01-05  51.490631   
182124  200801BS70005            Slight    2020-01-03  51.509712   
...               ...               ...           ...        ...   
352694       2.01E+12            Slight    2020-11-01  55.267100   
352695       2.01E+12            Slight    2020-11-02  54.985218   
352696       2.01E+12            Slight    2020-11-06  55.126057   
352697       2.01E+12           Serious    2020-11-04  55.243971   
352702       2.01E+12            Slight    2020-12-06  54.983911   

              Light_Conditions           District Area  Longitude  \
182120   Darkness - lights lit  Kensington and Chelsea  -0.179406   
182121   Darkness - lights lit  Kensington and Chelsea  -0.176356   
182122                Daylight  Kensington and Chelsea  -0.197405   
182123   Darkness - lights lit  Kensington and Chelsea  -0.185779   
182124   Darkness - lights lit  Kensington and Chelsea  -0.192659   
...                        ...                     ...        ...   
352694  Darkness - no lighting   Dumfries and Galloway  -2.959324   
352695                Daylight   Dumfries and Galloway  -3.278743   
352696  Darkness - no lighting   Dumfries and Galloway  -3.344404   
352697  Darkness - no lighting   Dumfries and Galloway  -3.461163   
352702                Daylight   Dumfries and Galloway  -3.195094   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
182120                     1                   2                     Dry   
182121                     1                   2             Wet or damp   
182122                     1                   1                     Dry   
182123                     1                   2                     Dry   
182124                     3                   1                     Dry   
...                      ...                 ...                     ...   
352694                     1                   1            Frost or ice   
352695                     1                   1             Wet or damp   
352696                     1                   1             Wet or damp   
352697                     2                   2                     Dry   
352702                     1                   1            Frost or ice   

                 Road_Type Urban_or_Rural_Area  Weather_Conditions  \
182120  Single carriageway               Urban  Fine no high winds   
182121  Single carriageway               Urban  Fine no high winds   
182122  Single carriageway               Urban  Fine no high winds   
182123  Single carriageway               Urban  Fine no high winds   
182124  Single carriageway               Urban  Fine no high winds   
...                    ...                 ...                 ...   
352694  Single carriageway               Rural  Fine no high winds   
352695  Single carriageway               Rural  Fine no high winds   
352696  Single carriageway               Rural  Fine no high winds   
352697  Single carriageway               Rural  Fine no high winds   
352702  Single carriageway               Rural  Fine no high winds   

                      Vehicle_Type    Year  Month   Day  DayOfWeek  
182120                         Car  2020.0    1.0   2.0        3.0  
182121       Motorcycle over 500cc  2020.0    1.0   6.0        0.0  
182122                         Car  2020.0    1.0  12.0        6.0  
182123                         Car  2020.0    1.0   5.0        6.0  
182124                         Car  2020.0    1.0   3.0        4.0  
...                            ...     ...    ...   ...        ...  
352694                         Car  2020.0   11.0   1.0        6.0  
352695  Motorcycle 125cc and under  2020.0   11.0   2.0        0

<h1> The Accidents decrease from 2019 with total of 71,867 accidents to 2020 accidents with total of 70,163 accidents. The decrease of accidents is -2.37%<h1>

<h1> Insight No.10 </h1>

In [59]:
acc1 = accident[(accident['Accident_Severity'] == 'Fatal') & (accident['Urban_or_Rural_Area'] == 'Rural')]

In [60]:
acc1

Index Accident_Severity Accident Date   Latitude  \
7542    200701KD65706             Fatal           NaT  51.510091   
8847    200701KG68848             Fatal           NaT  51.526004   
15926   200701TC00047             Fatal           NaT  51.500961   
15985   200701TC00116             Fatal           NaT  51.546020   
16009   200701TD00004             Fatal           NaT  51.409167   
...               ...               ...           ...        ...   
660470  201091NH00529             Fatal           NaT  58.475726   
660519  201091NJ01533             Fatal    2022-02-02  58.495554   
660565  201091NJ11329             Fatal           NaT  57.871514   
660581  201091NJ14695             Fatal           NaT  58.445475   
660627  201091NL00060             Fatal    2022-01-06  60.241195   

             Light_Conditions         District Area  Longitude  \
7542                 Daylight              Havering   0.196036   
8847                 Daylight  Barking and Dagenham   0.128033   
15926   Darkness - lights lit              Havering   0.217643   
15985                Daylight              Havering   0.231661   
16009   Darkness - lights lit               Bromley   0.143741   
...                       ...                   ...        ...   
660470               Daylight         Western Isles  -6.278583   
660519               Daylight              Highland  -4.451628   
660565               Daylight              Highland  -4.243081   
660581  Darkness - lights lit              Highland  -3.065535   
660627               Daylight      Shetland Islands  -1.409761   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
7542                       1                   2                     Dry   
8847                       1                   1             Wet or damp   
15926                      1                   2                     Dry   
15985                      2                   2                     Dry   
16009                      1                   1                     Dry   
...                      ...                 ...                     ...   
660470                     1                   1             Wet or damp   
660519                     1                   1                    Snow   
660565                     1                   1                     Dry   
660581                     1                   1             Wet or damp   
660627                     3                   1                    Snow   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
7542      Dual carriageway               Rural     Fine no high winds   
8847    Single carriageway               Rural                  Other   
15926     Dual carriageway               Rural     Fine no high winds   
15985   Single carriageway               Rural     Fine no high winds   
16009     Dual carriageway               Rural     Fine no high winds   
...                    ...                 ...                    ...   
660470  Single carriageway               Rural      Fine + high winds   
660519  Single carriageway               Rural  Snowing no high winds   
660565  Single carriageway               Rural     Fine no high winds   
660581  Single carriageway               Rural     Fine no high winds   
660627  Single carriageway               Rural     Fine no high winds   

                               Vehicle_Type    Year  Month  Day  DayOfWeek  
7542                                    Car     NaN    NaN  NaN        NaN  
8847                                    Car     NaN    NaN  NaN        NaN  
15926                                   Car     NaN    NaN  NaN        NaN  
15985                                   Car     NaN    NaN  NaN        NaN  
16009            Motorcycle 125cc and under     NaN    NaN  NaN        NaN  
...                                     ...     ...    ...  ...        ...  
660470                                  Car     NaN    NaN  NaN        NaN  


<h1>There are 5,601 Fatal Accidents happened in Rural Area.</h1>

<h1>Insight No.11</h1>

<h1> What year has the Highest Accidents Recorded in United Kingdom?</h1>

In [80]:
accident['Year'].value_counts()


Year
2019.0    71867
2020.0    70163
2021.0    66172
2022.0    56805
Name: count, dtype: int64

In [81]:
accident['Road_Surface_Conditions'].value_counts()

Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

<h1>2019 has the Highest Accidents recorded with over 71,867 accidents and most of the Accidents in United Kingdom happened in Dry Surface that contributes 68% to all Accidents.</h1>

<h1>Insight No.12</h1>

In [89]:
acc_conditions = accident[(accident['Light_Conditions'] == 'Daylight') & (accident['Road_Surface_Conditions'] == 'Dry')]

In [90]:
acc_conditions

Index Accident_Severity Accident Date   Latitude  \
3       200701BS66128           Serious           NaT  51.495478   
5       200701BS67159           Serious           NaT  51.497750   
6       200701BS67207           Serious    2019-09-05  51.501405   
9       200701BS67543            Slight           NaT  51.484539   
11      200701BS67747             Fatal    2019-11-02  51.499127   
...               ...               ...           ...        ...   
660638  201091NL04333            Slight    2022-08-06  60.306795   
660639  201091NL04383            Slight    2022-08-08  59.924538   
660645  201091NL05842           Serious           NaT  60.162660   
660648  201091NL06327            Slight    2022-11-12  60.157151   
660674  201091NM01760            Slight           NaT  57.374005   

       Light_Conditions           District Area  Longitude  \
3              Daylight  Kensington and Chelsea  -0.202731   
5              Daylight  Kensington and Chelsea  -0.192556   
6              Daylight  Kensington and Chelsea  -0.161577   
9              Daylight  Kensington and Chelsea  -0.175651   
11             Daylight  Kensington and Chelsea  -0.159652   
...                 ...                     ...        ...   
660638         Daylight        Shetland Islands  -1.242994   
660639         Daylight        Shetland Islands  -1.290893   
660645         Daylight        Shetland Islands  -1.165232   
660648         Daylight        Shetland Islands  -1.146277   
660674         Daylight                Highland  -3.467828   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
3                          1                   4                     Dry   
5                          2                   3                     Dry   
6                          1                   2                     Dry   
9                          1                   2                     Dry   
11                         1                   2                     Dry   
...                      ...                 ...                     ...   
660638                     2                   1                     Dry   
660639                     4                   2                     Dry   
660645                     1                   1                     Dry   
660648                     1                   1                     Dry   
660674                     2                   1                     Dry   

                 Road_Type Urban_or_Rural_Area  Weather_Conditions  \
3       Single carriageway               Urban  Fine no high winds   
5       Single carriageway               Urban  Fine no high winds   
6         Dual carriageway               Urban  Fine no high winds   
9       Single carriageway               Urban  Fine no high winds   
11      Single carriageway               Urban  Fine no high winds   
...                    ...                 ...                 ...   
660638  Single carriageway               Rural  Fine no high winds   
660639  Single carriageway               Rural  Fine no high winds   
660645  Single carriageway               Rural   Fine + high winds   
660648  Single carriageway               Rural  Fine no high winds   
660674  Single carriageway               Rural  Fine no high winds   

                                 Vehicle_Type    Year  Month   Day  DayOfWeek  
3        Bus or coach (17 or more pass seats)     NaN    NaN   NaN        NaN  
5                                         Car     NaN    NaN   NaN        NaN  
6         Van / Goods 3.5 tonnes mgw or under  2019.0    9.0   5.0        3.0  
9       Motorcycle over 125cc and up to 500cc     NaN    NaN   NaN        NaN  
11       Bus or coach (17 or more pass seats)  2019.0   11.0   2.0        5.0  
...                                       ...     ...    ...   ...        ...  
660638                                    Car  2022.0    8.0   6.0        5.0  
660639                                    Car  2022.0    8.0

In [91]:
accident_districtroad = accident[(acc_conditions['District Area'] == 'Birmingham') & (accident['Road_Type'] == 'Single carriageway')]

In [92]:
accident_districtroad

Index Accident_Severity Accident Date   Latitude  \
68521   200720D003802           Serious    2019-01-03  52.563201   
68524   200720D006001            Slight    2019-01-04  52.487709   
68538   200720D015003            Slight           NaT  52.494891   
68545   200720D017001            Slight    2019-01-10  52.491418   
68552   200720D018103            Slight    2019-01-08  52.473614   
...               ...               ...           ...        ...   
583514  201020W054720            Slight           NaT  52.502089   
583519  201020W055180            Slight           NaT  52.548094   
583525  201020W055950            Slight           NaT  52.505335   
583552  201020W503290            Slight           NaT  52.488673   
583556  201020W509790            Slight    2022-08-04  52.494991   

       Light_Conditions District Area  Longitude  Number_of_Casualties  \
68521          Daylight    Birmingham  -1.822793                     1   
68524          Daylight    Birmingham  -1.840032                     3   
68538          Daylight    Birmingham  -1.777995                     1   
68545          Daylight    Birmingham  -1.858724                     1   
68552          Daylight    Birmingham  -1.855247                     1   
...                 ...           ...        ...                   ...   
583514         Daylight    Birmingham  -1.928371                     1   
583519         Daylight    Birmingham  -1.895705                     1   
583525         Daylight    Birmingham  -1.945014                     1   
583552         Daylight    Birmingham  -1.898938                     2   
583556         Daylight    Birmingham  -1.935010                     1   

        Number_of_Vehicles Road_Surface_Conditions           Road_Type  \
68521                    1                     Dry  Single carriageway   
68524                    2                     Dry  Single carriageway   
68538                    2                     Dry  Single carriageway   
68545                    1                     Dry  Single carriageway   
68552                    2                     Dry  Single carriageway   
...                    ...                     ...                 ...   
583514                   2                     Dry  Single carriageway   
583519                   2                     Dry  Single carriageway   
583525                   2                     Dry  Single carriageway   
583552                   2                     Dry  Single carriageway   
583556                   2                     Dry  Single carriageway   

       Urban_or_Rural_Area  Weather_Conditions                   Vehicle_Type  \
68521                Urban  Fine no high winds                            Car   
68524                Urban               Other                            Car   
68538                Urban  Fine no high winds  Goods 7.5 tonnes mgw and over   
68545                Urban  Fine no high winds  Goods 7.5 tonnes mgw and over   
68552                Urban                 NaN          Motorcycle over 500cc   
...                    ...                 ...                            ...   
583514               Urban  Fine no high winds                            Car   
583519               Urban  Fine no high winds                            Car   
583525               Urban  Fine no high winds                            Car   
583552               Urban                 NaN                            Car   
583556               Urban  Fine no high winds                            Car   

          Year  Month   Day  DayOfWeek  
68521   2019.0    1.0   3.0        3.0  
68524   2019.0    1.0   4.0        4.0  
68538      NaN    NaN   NaN        NaN  
68545   2019.0    1.0  10.0        3.0  
68552   2019.0    1.0   8.0        1.0  
...        ...    ...   ...        ...  
583514     NaN    NaN   NaN        NaN  
583519     NaN    NaN   NaN        NaN  
583525     NaN    NaN   NaN        NaN  
583552     NaN    NaN   NaN        NaN

<h1> There are 356,085 accidents happened in Daylight with Dry Surface. </h1>

<h1> There are 4,952 accidents happened in Birmingham with Single Carriageway Road Type. </h1>

<h1>Insight No.13</h1>

In [109]:
accident_hotspots = accident['District Area'.value_counts().reset_index()]
accident_hotspots.columns = ['District Area', 'Accident Count']

AttributeError: 'str' object has no attribute 'value_counts'

In [110]:
plt.figure(figsize=(12, 6))
sns.barplot(x='Accident Count', y='District Area', data=accident_hotspots.head(10), palette='Reds')
plt.xlabel('Number of Accidents')
plt.ylabel('District Area')
plt.title('Top 10 Most Affected Locations by Accidents')
plt.show()

NameError: name 'accident_hotspots' is not defined

<Figure size 1200x600 with 0 Axes>

<h1>Insight No.14</h1>

In [112]:
acc_conditions1 = accident[(accident['Accident_Severity'] == 'Serious') & (accident['Road_Surface_Conditions'] == 'Dry')]

In [113]:
acc_conditions1

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
2       200701BS66127           Serious           NaT  51.517715   
3       200701BS66128           Serious           NaT  51.495478   
5       200701BS67159           Serious           NaT  51.497750   
6       200701BS67207           Serious    2019-09-05  51.501405   
...               ...               ...           ...        ...   
660609  201091NK02972           Serious           NaT  58.961359   
660613  201091NK03455           Serious           NaT  58.982692   
660619  201091NK04839           Serious           NaT  58.977273   
660629  201091NL00686           Serious    2022-02-11  57.687933   
660645  201091NL05842           Serious           NaT  60.162660   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
5                          Daylight  Kensington and Chelsea  -0.192556   
6                          Daylight  Kensington and Chelsea  -0.161577   
...                             ...                     ...        ...   
660609                     Daylight          Orkney Islands  -2.928920   
660613                     Daylight          Orkney Islands  -2.959245   
660619                     Daylight          Orkney Islands  -2.963443   
660629                     Daylight        Shetland Islands  -4.176860   
660645                     Daylight        Shetland Islands  -1.165232   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
5                          2                   3                     Dry   
6                          1                   2                     Dry   
...                      ...                 ...                     ...   
660609                     1                   1                     Dry   
660613                     1                   1                     Dry   
660619                     1                   2                     Dry   
660629                     1                   2                     Dry   
660645                     1                   1                     Dry   

                 Road_Type Urban_or_Rural_Area  Weather_Conditions  \
0       Single carriageway               Urban  Fine no high winds   
2                      NaN               Urban                 NaN   
3       Single carriageway               Urban  Fine no high winds   
5       Single carriageway               Urban  Fine no high winds   
6         Dual carriageway               Urban  Fine no high winds   
...                    ...                 ...                 ...   
660609  Single carriageway               Rural  Fine no high winds   
660613      One way street               Rural  Fine no high winds   
660619  Single carriageway               Rural  Fine no high winds   
660629  Single carriageway               Rural   Fine + high winds   
660645  Single carriageway               Rural   Fine + high winds   

                                Vehicle_Type    Year  Month   Day  DayOfWeek  
0                                        Car  2019.0    6.0   5.0        2.0  
2                      Taxi/Private hire car     NaN    NaN   NaN        NaN  
3       Bus or coach (17 or more pass seats)     NaN    NaN   NaN        NaN  
5                                        Car     NaN    NaN   NaN        NaN  
6        Van / Goods 3.5 tonnes mgw or under  2019.0    9.0   5.0        3.0  
...                                      ...     ...    ...   ...        ...  
660

<h1> There are 61,638 accidents happened in Dry Surface with Serious Accident Severity</h1>

<h1>Insight No.15</h1>

<h1>How many accidents with Fine + high winds condition in Rural Area?</h1>

In [115]:
acc_conditions2 = accident[(accident['Weather_Conditions'] == 'Fine + high winds') & (accident['Urban_or_Rural_Area'] == 'Rural')]

In [116]:
acc_conditions2

Index Accident_Severity Accident Date   Latitude  \
7159    200701KD50018           Serious           NaT  51.525255   
7425    200701KD62822            Slight    2019-05-12  51.512375   
12299   200701PY65398           Serious    2019-07-06  51.350693   
14273   200701RY10077            Slight    2019-03-05  51.445413   
15845   200701TB00016           Serious           NaT  51.629002   
...               ...               ...           ...        ...   
660629  201091NL00686           Serious    2022-02-11  57.687933   
660637  201091NL03904            Slight           NaT  60.233118   
660645  201091NL05842           Serious           NaT  60.162660   
660647  201091NL06131            Slight    2022-11-03  60.195539   
660665  201091NM00758            Slight           NaT  56.884448   

              Light_Conditions     District Area  Longitude  \
7159    Darkness - no lighting          Havering   0.231222   
7425                  Daylight          Havering   0.184472   
12299                 Daylight           Bromley   0.129952   
14273                 Daylight            Bexley   0.158247   
15845                 Daylight            Harrow  -0.329520   
...                        ...               ...        ...   
660629                Daylight  Shetland Islands  -4.176860   
660637                Daylight  Shetland Islands  -1.234763   
660645                Daylight  Shetland Islands  -1.165232   
660647                Daylight  Shetland Islands  -1.258363   
660665                Daylight          Highland  -4.252047   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
7159                       1                   1             Wet or damp   
7425                       1                   1             Wet or damp   
12299                      3                   2                     Dry   
14273                      1                   2                     Dry   
15845                      2                   2                     Dry   
...                      ...                 ...                     ...   
660629                     1                   2                     Dry   
660637                     1                   2                     Dry   
660645                     1                   1                     Dry   
660647                     1                   1             Wet or damp   
660665                     1                   2                    Snow   

                 Road_Type Urban_or_Rural_Area Weather_Conditions  \
7159    Single carriageway               Rural  Fine + high winds   
7425      Dual carriageway               Rural  Fine + high winds   
12299   Single carriageway               Rural  Fine + high winds   
14273     Dual carriageway               Rural  Fine + high winds   
15845   Single carriageway               Rural  Fine + high winds   
...                    ...                 ...                ...   
660629  Single carriageway               Rural  Fine + high winds   
660637  Single carriageway               Rural  Fine + high winds   
660645  Single carriageway               Rural  Fine + high winds   
660647  Single carriageway               Rural  Fine + high winds   
660665  Single carriageway               Rural  Fine + high winds   

                                Vehicle_Type    Year  Month   Day  DayOfWeek  
7159                                     Car     NaN    NaN   NaN        NaN  
7425                                     Car  2019.0    5.0  12.0        6.0  
12299   Bus or coach (17 or more pass seats)  2019.0    7.0   6.0        5.0  
14273                  Motorcycle over 500cc  2019.0    3.0   5.0        1.0  
15845                                    Car     NaN    NaN   NaN        NaN  
...                                      ...     ...    ...   ...        ...  
660629                                   Car  2022.0    2.0  11.0        4.0  
660637                                   Car     NaN    NaN   NaN   

<h1>There are 3,993 accidents with Fine + high winds in Rural Area.</h1>

<h1>Insight No.16</h1>

<h1>How many accidents happen in Darkness -no lightning in Motorcycle? </h1>

In [ ]:
acc_conditions3 = accident[(accident['Light_Conditions'] == 'Daylight') & (accident['Vehicle_Type'] == 'Car')]